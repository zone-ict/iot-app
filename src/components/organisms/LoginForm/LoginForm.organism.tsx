import { FormikConfig, useFormik } from 'formik';
import React, { useMemo } from 'react';
import InputField from '../../molecules/InputField/InputField.molecule';

type LoginSchema = {
  username: string;
  password: string;
};

interface Props {
  formConfig: FormikConfig<LoginSchema>;
  isLoading: boolean;
}

export default function LoginForm({ formConfig, isLoading }: Props) {
  const { errors, values, handleChange, handleBlur, handleSubmit, isValid } = useFormik(formConfig);

  const disabled = useMemo(() => !isValid || isLoading, [isLoading, isValid]);

  return (
    <form onSubmit={handleSubmit}>
      <InputField
        onChange={handleChange}
        onBlur={handleBlur}
        error={errors.username}
        value={values.username}
        name="username"
      />
      <InputField
        onChange={handleChange}
        onBlur={handleBlur}
        error={errors.password}
        type="password"
        name="password"
      />
      <button disabled={disabled} type="submit">
        {isLoading ? 'Loading...' : 'Submit'}
      </button>
    </form>
  );
}
