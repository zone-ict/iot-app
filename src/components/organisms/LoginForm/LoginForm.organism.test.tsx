import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm.organism';

const testConfig = {
  initialValues: {
    username: '',
    password: '',
  },
  onSubmit: () => {},
};

describe('Login form component', () => {
  it('should be able to render correctly with false loading prop', () => {
    const tree = renderer.create(<LoginForm isLoading={false} formConfig={testConfig} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to render correctly with true loading prop', () => {
    const tree = renderer.create(<LoginForm isLoading formConfig={testConfig} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
