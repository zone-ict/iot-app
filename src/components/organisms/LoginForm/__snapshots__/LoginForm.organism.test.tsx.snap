// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component should be able to render correctly with false loading prop 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <h6 />
    <input
      className="css-1fq7r63"
      name="username"
      onBlur={[Function]}
      onChange={[Function]}
      value=""
    />
  </div>
  <div>
    <h6 />
    <input
      className="css-1fq7r63"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
    />
  </div>
  <button
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`Login form component should be able to render correctly with true loading prop 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <h6 />
    <input
      className="css-1fq7r63"
      name="username"
      onBlur={[Function]}
      onChange={[Function]}
      value=""
    />
  </div>
  <div>
    <h6 />
    <input
      className="css-1fq7r63"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Loading...
  </button>
</form>
`;
